src/ResourceManagement/Compute/Domain/InterfaceImpl/VirtualMachineImpl.cs

```diff
@@ -645,6 +645,16 @@ VirtualMachine.Update.IUpdate VirtualMachine.Update.IUpdate.WithSize(VirtualMach
             return this.WithSize(size);
         }
 
+        /// <summary>
+        /// Specifies the license-type for the virtual machine.
+        /// </summary>
+        /// <param name="licenseType">The license-type that should be used for the virtual machine.</param>
+        /// <return>The next stage of the definition.</return>
+        VirtualMachine.Update.IUpdate VirtualMachine.Update.IUpdate.WithLicenseType(string licenseType)
+        {
+            return this.WithLicenseType(licenseType) as VirtualMachine.Update.IUpdate;
```

Comment:
According to swagger spec description, this value cannot be updated: https://github.com/Azure/azure-rest-api-specs/blob/c8516b913b3c420f38e036c3e2b8dbf1550cb089/specification/compute/resource-manager/Microsoft.Compute/stable/2018-04-01/compute.json#L5796

Reply:
I pretty sure the update also worked when I tested this, but since it is already a couple of weeks ago I am not 100% sure about it anymore. On the other hand in the Azure Portal you can enable and disable "Azure Hybrid Benefit" as many times as you want.  

So I tested it again. I am able to set and empty the license-type multiple times for the same VM. However, I did not check if I could set another license-type after I used: "Windows_Server".
But I guess when this is not supported the server will return an exception.